### PR TITLE
ETH - Fix tx history

### PR DESCRIPTION
**Pull request by huazhouwang**
## What does this implement/fix? Explain your changes.
1. 修复 ETH 无法获取交易历史问题，返回数据格式跟 BTC 完全一样 
2. 在这基础上，感觉原 electrum 项目缺少分层结构化的思想。我根据之前的实践，在 electrum_gui 下新建了 common 目录，目前封装好了 dataclass，restful request 并实现 TrezorETH 和 Etherscan 这两个浏览器实例来获取交易历史等链上数据，后续还会继续扩展。

## Does this close any currently open issues?  
If it fixes a bug or resolves a feature request, be sure to link to that issue.
Nope

## Pull request type
 
_Put an `x` in the boxes that apply_
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Where has this been tested?
1. 获取 ETH 交易记录

## Any other comments?
后续我会继续扩展 electrum_gui.common， 最终目标是达到添加新币只需要实现指定几个接口即可

### CODE REVIEW

**Review by shuaiLiWang — APPROVED** [review]

File: electrum/pywalib.py [hunk 1]
```diff
@@ -447,15 +453,64 @@ def get_balance(wallet_address, contract=None):
...
+            servers = {list(i.keys())[0]: list(i.values())[0] for i in cls.tx_list_server}
+            explorer = (
+                TrezorETH(
+                    servers.get("onekey-mainnet")
+                    or servers.get("trezor-mainnet")
+                )
+                if cls.chain_type == "mainnet"
+                else Etherscan(servers.get("etherscan-testnet"), ETHERSCAN_API_KEY)
+            )
+            setattr(cls, cache_name, explorer)
```
[the remark this answers is not in the document]
Comment (huazhouwang) [reply electrum/pywalib.py:471]: 嗯，问题已记录，后续会加

File: electrum_gui/android/console.py [hunk 2]
```diff
@@ -3230,7 +3228,7 @@ def filter_wallet(self):
                 coin = wallet.wallet_type[0:3]
                 if self.coins.__contains__(wallet.wallet_type[0:3]):
                     address = wallet.get_addresses()[0]
-                    tx_list = PyWalib.get_transaction_history(address, recovery=True)
+                    tx_list = self.get_eth_tx_list(address)
```
[the remark this answers is not in the document]
Comment (huazhouwang) [reply electrum_gui/android/console.py:3231]: 了解，策略后续会加

**Comment by huazhouwang** [comment]
关于 Django 代码那块，目前看它的版权描述，没有针对只取两个方法的情况，https://github.com/django/django/blob/master/LICENSE

File: electrum_gui/common/basic/functional/decimal_encoder.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import json
+from decimal import Decimal
+
+
+class DecimalEncoder(json.JSONEncoder):
```
Comment (danielsocials) [inline electrum_gui/common/basic/functional/decimal_encoder.py:5]: 删除,用electrum/util.py:418:def json_encode(obj)

File: electrum_gui/common/basic/functional/text.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import datetime
```
Comment (danielsocials) [inline electrum_gui/common/basic/functional/text.py:1]: 在文件头添加 
Copyright (c) Django Software Foundation and individual contributors.
All rights reserved.

**Review by danielsocials — CHANGES_REQUESTED** [review]

**Pull request merged**